### PR TITLE
[v6r21]  fix fts duplicate transfer (and fix Script.py)

### PR DESCRIPTION
**Pull request by chaen**
So, there is a bug in the logic in the new FTS system, which might result in duplicate transfers. This is of course not good. This PR fixes it a priori, but unfortunately requires a table change.

The duplicate transfers happen like that:
* Operation has two files File1 and File2.
* Job1 is submitted for File1 and File2.
* File1 fails, File2 is still ongoing.
* We submit Job2 for File1.
* Job1 is monitored again, and we update again File1 to failed (because it is so in Job1)
* A Job3 would be created for File1, despite Job2 still running on it.

The idea of the fix is as follow:
* we add an `ftsGUID` attribute to the FTSFile, initialized to None.
* when submitting a job for that file, we set the ftsGUID to the value of the job we just submitted.  
* when the job is monitored, if the transfer for that file is in an fts final state (Done, Cancel, Failed), we set the ftsGUID to None
* when the status of files are updated after the monitoring of the job, we specify the GUID of the job in the query, such that only the files with this specific GUID are updated. 

@andresailer  please check it carefully. And if you still have your setup to test it, I would love that ! 
@phicharp you may have comments as well.

If we are happy with this solution, I will provide instruction and script to update the database.

PS: I also added a fix for Script.py

BEGINRELEASENOTES

*DMS
FIX: Fixes FTS3 duplicate transfers

*Core
FIX: allow Script.py to accommodate specific test calls with pytest

ENDRELEASENOTES

### CODE REVIEW

File: DataManagementSystem/Agent/FTS3Agent.py [hunk 1]
```diff
@@ -213,7 +213,9 @@ def _monitorJob(self, ftsJob):
       # { fileID : { Status, Error } }
       filesStatus = res['Value']
 
-      res = self.fts3db.updateFileStatus(filesStatus)
+      # Specify the job ftsGUID to make sure we do not overwrite
+      # status of files already taken by newer jobs
+      res = self.fts3db.updateFileStatus(filesStatus, ftsGUID = ftsJob.ftsGUID)
```
[the remark this answers is not in the document]
Comment (chaen) [reply DataManagementSystem/Agent/FTS3Agent.py]: Strange ! Why did pycodestyle not complain about that ?

[the remark this answers is not in the document]
Comment (andresailer) [reply DataManagementSystem/Agent/FTS3Agent.py]: Maybe I was looking at an older version, I started looking long time ago, always forgot to submit the comments...

**Comment by chaen** [comment]
I corrected all the spelling mistakes

**Comment by chaen** [comment]
@atsareg @fstagni This will require a change in the DB and a procedure to perform it. 
I propose that Andrei creates the release, but do not announce it. We create an LHCbDIRAC release on top of it, and we polish the migration procedure (easy in principle). And once we are happy, we announce the release

**Pull request merged**